### PR TITLE
React validation migration

### LINKED ISSUE #13774: Warning from Reactstrap validation
##### **Overview of the issue**

I saw this when testing webflux-couchbase today.

```
[INFO] --- frontend-maven-plugin:1.11.0:npm (npm install) @ sample-couchbase-no-cache ---
[INFO] Running 'npm install' in /Users/mraible/dev/webflux-couchbase
[INFO] npm WARN EBADENGINE Unsupported engine {
[INFO] npm WARN EBADENGINE   package: 'availity-reactstrap-validation@2.7.0',
[INFO] npm WARN EBADENGINE   required: { node: '>= 5.0.0', npm: '^3.0.0' },
[INFO] npm WARN EBADENGINE   current: { node: 'v14.15.0', npm: '7.4.3' }
[INFO] npm WARN EBADENGINE }
[INFO]
[INFO] up to date, audited 2450 packages in 2s
```

Not sure if it's something we should be concerned about.

##### **Motivation for or Use Case**

No one likes to see errors or warnings when building their app.

##### **Reproduce the error**

Create a project with this `.yo-rc.json`. You'll see the warning as part of `npm install`.

```json
{
  "generator-jhipster": {
    "baseName": "sampleCouchbaseNoCache",
    "packageName": "tech.jhipster.sample",
    "packageFolder": "tech/jhipster/sample",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "websocket": false,
    "databaseType": "couchbase",
    "devDatabaseType": "couchbase",
    "prodDatabaseType": "couchbase",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "clientPackageManager": "npm",
    "applicationType": "monolith",
    "testFrameworks": ["gatling", "cypress"],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en", "fr"],
    "creationTimestamp": 1596513172471,
    "jwtSecretKey": "ZjY4MTM4YjI5YzMwZjhjYjI2OTNkNTRjMWQ5Y2Q0Y2YwOWNmZTE2NzRmYzU3NTMwM2NjOTE3MTllOTM3MWRkMzcyYTljMjVmNmQ0Y2MxOTUzODc0MDhhMTlkMDIxMzI2YzQzZDM2ZDE3MmQ3NjVkODk3OTVmYzljYTQyZDNmMTQ=",
    "blueprints": [],
    "otherModules": [],
    "jhipsterVersion": "7.0.0-beta.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "pages": [],
    "clientTheme": "none",
    "clientThemeVariant": "",
    "withAdminUi": true,
    "reactive": true,
    "entities": [
      "DocumentBankAccount",
      "FieldTestEntity",
      "FieldTestInfiniteScrollEntity",
      "FieldTestMapstructAndServiceClassEntity",
      "FieldTestPaginationEntity",
      "FieldTestServiceClassAndJpaFilteringEntity",
      "FieldTestServiceImplEntity",
      "EntityWithDTO",
      "EntityWithServiceClassAndPagination",
      "EntityWithServiceImplAndPagination",
      "EntityWithServiceImplAndDTO",
      "EntityWithPaginationAndDTO",
      "EntityWithServiceClassPaginationAndDTO",
      "EntityWithServiceImplPaginationAndDTO",
      "Division",
      "Place",
      "EmbeddedOperation"
    ]
  }
}
```

##### **JHipster Version(s)**

`fix-couchbase` branch.

### LINKED ISSUE #15127: If you create a new react jhipster project it and add entities it uses availity-reactstrap-validation which is deprecated
##### **Overview of the issue**

If you create a new react jhipster project it uses availity-reactstrap-validation which is deprecated. If we create new projects, is the only workaround to migrate all the react code manually from availity-reactstrap-validation to availity-react?

##### **Motivation for or Use Case**

Doesn't feel like a great start that I am using deprecated libraries if I create entities in a new react Jhipster app. 
##### **Reproduce the error**

1. Create JHipster app with react
2. Add entitiy
3. See availity-reactstrap-validation usage. 

##### **Suggest a Fix**

Upgrade jhipster to use availity-react or other. 

##### **JHipster Version(s)**

7.0.1

##### **JHipster configuration**

INFO! Using JHipster version installed locally in current project's node_modules
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
fengshui-manager@0.0.1-SNAPSHOT /Users/davidcarter/work/fengshuiManager
└── generator-jhipster@7.0.1

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "fengshuiManager",
    "jhipsterVersion": "7.0.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "pages": [],
    "creationTimestamp": 1620227346794,
    "serviceDiscoveryType": "eureka",
    "reactive": false,
    "authenticationType": "jwt",
    "packageName": "com.lime.fsm",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "mongodb",
    "buildTool": "gradle",
    "serverSideOptions": ["searchEngine:elasticsearch"],
    "websocket": false,
    "searchEngine": "elasticsearch",
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": false,
    "nativeLanguage": "en",
    "packageFolder": "com/lime/fsm",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "devDatabaseType": "mongodb",
    "prodDatabaseType": "mongodb",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["en", "fr"],
    "lastLiquibaseTimestamp": 1620753169000,
    "entities": ["Partialresult", "CalcsheetResults", "Quiz"]
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Partialresult {
  emailAddress String required unique
  responseData Instant
  floorPlansUploaded Boolean
  houseImagesUploaded Boolean
  folderName String
}
entity CalcsheetResults {
  averageCompass String
  adjustedCompass String
  sittingAndFacing String
  outsideWaterVolume String
  fsConstructionCycle String
  totalSquareFootage String
  metalPounds String
  metalKg String
  heavyMetalKg String
}
entity Quiz {
  name String required
  emailAddress String required unique minlength(5) maxlength(50)
  houseLock String
  mountainEnergy String
  houseRemedies String
  metalRemedies String
  twoStoreyHouse String
  twoStoryHouse Boolean
  movingMetal String
  fireOrRedArea String
  redEarthArea String
}
paginate Partialresult, CalcsheetResults, Quiz with pagination
service Partialresult, CalcsheetResults, Quiz with serviceImpl

</pre>
</details>

##### **Environment and Tools**

openjdk version "15" 2020-09-15
OpenJDK Runtime Environment (build 15+36-1562)
OpenJDK 64-Bit Server VM (build 15+36-1562, mixed mode, sharing)

git version 2.30.1 (Apple Git-130)

node: v15.1.0

npm: 7.0.8

Docker version 20.10.6, build 370c289

docker-compose version 1.29.1, build c34c88b2

No change to package.json was detected. No package manager install will be executed.
Congratulations, JHipster execution is complete!

##### **Browsers and Operating System**

n/a but on a mac

- [x ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by deepu105**
Fix #15127 
Fix #13774 

Ooff. This ended up being way more work then I anticipated :(

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by deepu105** [comment]
@pascalgrimaud i'll check the failing builds next. Those errors looks weird though

**Comment by deepu105** [comment]
@jhipster/developers can someone review this plz

File: generators/client/templates/react/src/main/webapp/app/config/notification-middleware.spec.ts.ejs [hunk 1]
```diff
@@ -196,9 +239,33 @@ describe('Notification Middleware', () => {
...
     expect((toastify.toast as any).success.called).toEqual(false);
   });
+
+  it('should trigger an error toast message and return error for 400 response code', () => {
+    expect(store.dispatch(TITLE_ERROR).error.response.status).toEqual(400);
+    const toastMsg = (toastify.toast as any).error.getCall(0).args[0];
+    expect(toastMsg).toContain('Incorrect password');
+  });
+
+  it('should trigger an error toast message and return error for string in data', () => {
```
Comment (qmonmert) [inline generators/client/templates/react/src/main/webapp/app/config/notification-middleware.spec.ts.ejs:254]: `should trigger an error toast message and return error for string in data` is the same L254, L260 and L266

File: generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs [hunk 2]
```diff
@@ -39,57 +38,61 @@ export const PasswordResetFinishPage = (props: RouteComponentProps<{ key: string
...
     );
   };
 
   const successMessage = useAppSelector(state => state.passwordReset.successMessage);
 
 <%_ if (enableTranslation) { _%>
-  if (successMessage) {
-    toast.success(translate(successMessage));
-  }
+  useEffect(() => {
```
Comment (qmonmert) [inline generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs:85]:
```
  useEffect(() => {
    if (successMessage) {
      <%_ if (enableTranslation) { _%>
      toast.success(translate(successMessage));
      <%_ } else { _%>
      toast.success(successMessage);
      <%_ } _%>
    }
  }, [successMessage]);
```
simpler no? :)

File: generators/client/templates/react/src/main/webapp/app/config/notification-middleware.spec.ts.ejs (hunk 1 above)
Comment (deepu105) [reply generators/client/templates/react/src/main/webapp/app/config/notification-middleware.spec.ts.ejs:254]: good catch

File: generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs (hunk 2 above)
Comment (deepu105) [reply generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/finish/password-reset-finish.tsx.ejs:85]: definite. Shit what was i thinking :facepalm:

File: generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/init/password-reset-init.tsx.ejs [hunk 3]
```diff
@@ -35,21 +34,24 @@ export const PasswordResetInit = () => {
...
+    dispatch(handlePasswordResetInit(email));
   };
 
   const successMessage = useAppSelector(state => state.passwordReset.successMessage);
 
 <%_ if (enableTranslation) { _%>
-  if (successMessage) {
-    toast.success(translate(successMessage));
-  }
+  useEffect(() => {
```
Comment (qmonmert) [inline generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/init/password-reset-init.tsx.ejs:44]: same

Comment (deepu105) [reply generators/client/templates/react/src/main/webapp/app/modules/account/password-reset/init/password-reset-init.tsx.ejs:44]: ya will update all places

File: generators/client/templates/react/src/main/webapp/app/modules/account/password/password.tsx.ejs [hunk 4]
```diff
@@ -50,17 +49,21 @@ export const PasswordPage = () => {
   const errorMessage = useAppSelector(state => state.password.errorMessage);
 
 <%_ if (enableTranslation) { _%>
-  if (successMessage) {
-    toast.success(translate(successMessage));
-  } else if (errorMessage) {
-    toast.success(translate(errorMessage));
-  }
+  useEffect(() => {
```
Comment (qmonmert) [inline generators/client/templates/react/src/main/webapp/app/modules/account/password/password.tsx.ejs:52]: same

File: generators/client/templates/react/src/main/webapp/app/modules/account/register/register.tsx.ejs [hunk 5]
```diff
@@ -41,27 +40,30 @@ export const RegisterPage = () => {
...
 
   const updatePassword = event => setPassword(event.target.value);
 
   const successMessage = useAppSelector(state => state.register.successMessage);
 
 <%_ if (enableTranslation) { _%>
-  if (successMessage) {
-    toast.success(translate(successMessage));
-  }
+  useEffect(() => {
```
Comment (qmonmert) [inline generators/client/templates/react/src/main/webapp/app/modules/account/register/register.tsx.ejs:56]: same

File: generators/client/templates/react/src/main/webapp/app/modules/account/settings/settings.tsx.ejs [hunk 6]
```diff
@@ -41,26 +40,29 @@ export const SettingsPage = () => {
     };
   }, []);
 
-  const handleValidSubmit = (event, values) => {
+<%_ if (enableTranslation) { _%>
```
Comment (qmonmert) [inline generators/client/templates/react/src/main/webapp/app/modules/account/settings/settings.tsx.ejs:43]: same

**Review by qmonmert — APPROVED** [review]

**Pull request merged**